lib/kernel/sync.rb

```diff
@@ -29,7 +29,13 @@ def Sync(*arguments, **options, &block)
 			reactor = Async::Reactor.new
 			
 			begin
-				return reactor.run(*arguments, **options, finished: ::Async::Condition.new, &block).wait
+				options = {
+					finished: ::Async::Condition.new
+				}
+
+				options[:annotation] = annotation if annotation
+
+				return reactor.run(**options, &block).wait
```

Comment:
Can't you just pass `annotation: annotation`?

Reply:
No, then some tests won't pass.

Reply:
I tried again. The tests do work now with the simpler variant.